Client/Frontend/Browser/DataURIParser.swift

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+class DataURIParser {
+    let mediaType: String
+    let headers: [String]
+    let data: Data
+    
+    init(uri: String) throws {
+        if uri == "data:," {
+            mediaType = "text/plain"
+            headers = ["charset=US-ASCII"]
+            data = Data()
+            return
+        }
+        
+        if !uri.lowercased().hasPrefix("data:") {
+            throw "Invalid Data-URI"
+        }
+        
+        guard let infoSegment = uri.firstIndex(of: ",") else {
+            throw "Invalid Data-URI"
+        }
+        
+        let startSegment = uri.index(uri.startIndex, offsetBy: "data:".count)
+        self.headers = uri[startSegment..<infoSegment].split(separator: ";").map({ String($0) })
+        mediaType = headers.first ?? "text/plain"
+        
+        let dataSegment = uri.index(infoSegment, offsetBy: 1)
+        //data = try! Data(contentsOf: URL(string: uri)!)
+        data = Data(base64Encoded: String(uri[dataSegment..<uri.endIndex])) ?? Data()
```

Comment:
Do we really want to fallback to an empty data object?
Maybe throwing an error would be better, what do you think?